tests/Agent/IntegrationTests/ContainerIntegrationTests/Fixtures/ContainerTestFixtures.cs

```diff
@@ -0,0 +1,138 @@
+// Copyright 2020 New Relic, Inc. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System;
+using System.Threading.Tasks;
+using NewRelic.Agent.ContainerIntegrationTests.Applications;
+using NewRelic.Agent.IntegrationTestHelpers.RemoteServiceFixtures;
+
+namespace NewRelic.Agent.ContainerIntegrationTests.Fixtures;
+
+public abstract class ContainerTestFixtureBase : RemoteApplicationFixture
+{
+    private const string DotnetVersion = "7.0";
+
+    protected override int MaxTries => 1;
+
+    protected ContainerTestFixtureBase(string applicationDirectoryName, string distroTag, ContainerApplication.Architecture containerArchitecture, string dockerfile) :
+        base(new ContainerApplication(applicationDirectoryName, distroTag, containerArchitecture, DotnetVersion, dockerfile))
+    {
+    }
+
+    public virtual void ExerciseApplication()
+    {
+        var address = $"http://localhost:{Port}/weatherforecast";
+        GetAndAssertStatusCode(address, System.Net.HttpStatusCode.OK);
+    }
+
+    public void Delay(int seconds)
+    {
+        Task.Delay(TimeSpan.FromSeconds(seconds)).GetAwaiter().GetResult();
+    }
+}
+
+public class DebianX64ContainerTestFixture : ContainerTestFixtureBase
+{
+    private const string Dockerfile = "SmokeTestApp/Dockerfile";
+    private const string DockerComposeServiceName = "DebianX64ContainerTestApp";
```

Comment:
I'm just wondering if we need to rename these constants to better match what they represent since we do not have unique docker compose service names as part of these changes.

Reply:
This "name" goes into building the container name and image name. So it's not really referencing a specific service in `docker-compose` (there's only one now!) - it might be possible to derive the name without needing to have a const for it. It's still important that these names are unique among all tests, which definitely won't be enforced by using consts to define them. I'll take a look and see what I can come up with.